Mathlib/Topology/Category/TopCat/Sphere.lean

```diff
@@ -50,4 +59,32 @@ def diskBoundaryInclusion (n : ℕ) : ∂𝔻 n ⟶ 𝔻 n :=
         rw [isOpen_induced_iff, ← hst, ← hrs]
         tauto⟩ }
 
+/-- The inclusion `𝔹 n ⟶ 𝔻 n` of the interior of the `n`-disk. -/
+def ballInclusion (n : ℕ) : 𝔹 n ⟶ 𝔻 n :=
+  ofHom
+    { toFun := fun ⟨p, hp⟩ ↦ ⟨p, Metric.ball_subset_closedBall hp⟩
+      continuous_toFun := ⟨fun t ⟨s, ⟨r, hro, hrs⟩, hst⟩ ↦ by
+        rw [isOpen_induced_iff, ← hst, ← hrs]
+        tauto⟩ }
+
+instance {n : ℕ} : Mono (diskBoundaryInclusion n) := mono_iff_injective _ |>.mpr <| by
+  intro ⟨x, hx⟩ ⟨y, hy⟩ h
+  simp [diskBoundaryInclusion, disk] at h
+  cases h; rfl
+
+instance {n : ℕ} : Mono (ballInclusion n) := TopCat.mono_iff_injective _ |>.mpr <| by
+  intro ⟨x, hx⟩ ⟨y, hy⟩ h
+  simp [ballInclusion, disk] at h
+  cases h; rfl
+
+instance compact_disk (n : ℕ) : CompactSpace (𝔻 n) := by
+  convert Homeomorph.compactSpace Homeomorph.ulift.symm
+  rw [← isCompact_iff_compactSpace]
+  fapply ProperSpace.isCompact_closedBall
+
+instance compact_sphere (n : ℕ) : CompactSpace (∂𝔻 n) := by
+  convert Homeomorph.compactSpace Homeomorph.ulift.symm
+  rw [← isCompact_iff_compactSpace]
+  fapply isCompact_sphere
```

Comment:
This exists already as [Metric.sphere.compactSpace](https://leanprover-community.github.io/mathlib4_docs/Mathlib/Topology/MetricSpace/ProperSpace.html#Metric.sphere.compactSpace) (incidentally misnamed, but let's ignore that)
```suggestion
  infer_instance
```